### DISCUSSION TITLE
PrimeNG components A11y focus and focus-visible management

### DISCUSSION THREAD

**Discussion by The-Real-1** [primefaces/community#1476] [PrimeNG]
Hi,

Is there a way to differenciate the focus gained by your components depending on where the focus came from : e.g. mouse or tabulation ?
What I see in your code is that you add a class (e.g. p-focus) regardless on the event source.
How can we stylise the behaviour of the components if we need to handle differently focus and focus-visible ?

Thanks a lot.

**Comment by cetincakiroglu** [comment]
Hi,

We don't consider event source if the component receives focus, and most of the UI libraries do the same. You may use custom CSS to implement it.

**Comment by The-Real-1** [comment]
Hi,

Maybe I'm wrong but even if I create some selectors for the :focus-visible, they will never be applied based on what is made on PrimeNg components.
Another point is that CSS can't handle exclusives selectors (eg  :focus-visble and not :focus ) so I don't see how to handle it without diving deep inside PrimeNg components structure.
